server/handlers/logic.go

```diff
@@ -0,0 +1,89 @@
+package handlers
+
+import (
+	"io"
+
+	"github.com/Sirupsen/logrus"
+	ctxu "github.com/docker/distribution/context"
+	"golang.org/x/net/context"
+
+	"github.com/docker/notary/server/errors"
+	"github.com/docker/notary/server/snapshot"
+	"github.com/docker/notary/server/storage"
+	"github.com/docker/notary/server/timestamp"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+)
+
+func getRole(ctx context.Context, logger ctxu.Logger, w io.Writer, store storage.MetaStore, gun, role, checksum string) error {
+	var (
+		out []byte
+		err error
+	)
+	if checksum == "" {
+		// the timestamp and snapshot might be server signed so are
+		// handled specially
+		switch role {
+		case data.CanonicalTimestampRole, data.CanonicalSnapshotRole:
+			return getMaybeServerSigned(ctx, w, logger, store, gun, role)
+		}
+		out, err = store.GetCurrent(gun, role)
+	} else {
+		out, err = store.GetChecksum(gun, role, checksum)
+	}
+
+	if err != nil {
+		if _, ok := err.(storage.ErrNotFound); ok {
+			logrus.Errorf("404 GET %s:%s@%s, error: %v", gun, role, checksum, err)
+			return errors.ErrMetadataNotFound.WithDetail(nil)
+		}
+		logger.Error("500 GET")
+		return errors.ErrUnknown.WithDetail(err)
+	}
+	if out == nil {
+		logger.Error("404 GET")
+		return errors.ErrMetadataNotFound.WithDetail(nil)
+	}
+	w.Write(out)
+	logger.Debug("200 GET")
+
+	return nil
+}
+
+// getMaybeServerSigned writes the current snapshot or timestamp (based on the
+// role passed) to the provided writer or returns an error. In retrieving
+// the timestamp and snapshot, based on the keys held by the server, a new one
+// might be generated and signed due to expiry of the previous one or updates
+// to other roles.
```

Comment:
On repository initialization or key rotation from local to remote, the server provides back information about the keyID that should be added to the TUF metadata and signed by the client with the root key.

The question is: can the server provide more than the data for the timestamp and snapshot in that case, potentially having the client sign arbitrary TUF metadata?